File: bench/ary/ary.php
#!/usr/bin/php4 -f
<?php
/*
 $Id: ary.php,v 1.1 2004-05-28 06:39:25 bfulgham Exp $
 http://www.bagley.org/~doug/shootout/
*/
$n = ($argc == 2) ? $argv[1] : 1;
for ($i=0; $i<$n; $i++) {
    $X[$i] = $i + 1;
}
for ($k=0; $k<1000; $k++) {
    for ($i=$n-1; $i>=0; $i--) {
	$Y[$i] += $X[$i];
    }
}
$last = $n-1;
print "$Y[0] $Y[$last]\n";
?>

File: bench/ary/ary.lua
-- $Id: ary.lua,v 1.1 2004-05-19 18:09:15 bfulgham Exp $
-- http://www.bagley.org/~doug/shootout/
-- Author: Roberto Ierusalimschy

local n = tonumber((arg and arg[1]) or 1)

local x, y = {}, {}
for i=1,n do
  x[i] = i
end

for j=n,1,-1 do
  y[j] = x[j]
end

io.write(y[n-1], "\n")
